gnucash/gnome-utils/gnc-account-sel.c

```diff
@@ -144,33 +145,241 @@ combo_changed_cb (GNCAccountSel *gas, gpointer combo)
+                            c = g_utf8_next_char (c);
+                            item_offset_to_sep_char ++;
+                        }
+
+                        if (found && (item_offset_to_sep_char < prefix_len))
+                        {
+                            prefix_len = item_offset_to_sep_char;
+                            memset (gas->prefix, 0, BUFLEN);
+                            g_utf8_strncpy (gas->prefix, account_full_name, prefix_len);
+                        }
+                    }
+                }
+                g_free (lower_account_full_name);
+                g_free (account_full_name);
+                valid = gtk_tree_model_iter_next (GTK_TREE_MODEL(gas->store), &iter);
+            }
+            if (gas->prefix[0] == 0)
+                g_utf8_strncpy (gas->prefix, lower_entered_text, entered_len);
+
+            g_free (lower_entered_text);
+        }
+        g_free (normal_entered_text);
+    }
+
+    if (gas->prefix[0] != 0)
+    {
+        g_signal_handlers_block_by_func (editable, (gpointer) entry_insert_text_cb, user_data);
+        gtk_editable_delete_text (editable, 0, -1);
+        gtk_editable_set_position (editable, 0);
+        gtk_editable_insert_text (editable, gas->prefix, -1, position);
+        g_signal_handlers_unblock_by_func (editable, (gpointer) entry_insert_text_cb, user_data);
+        g_signal_stop_emission_by_name (editable, "insert_text");
+    }
+    g_free (entered_text);
+}
+
+static void
+update_entry_and_list (GNCAccountSel *gas)
+{
+    GtkEntry *entry = GTK_ENTRY(gtk_bin_get_child (GTK_BIN(gas->combo)));
+
+    g_signal_handlers_block_by_func (gas->combo, combo_changed_cb , gas);
+    gtk_entry_set_text (entry, "");
+    g_signal_handlers_unblock_by_func (gas->combo, combo_changed_cb , gas);
+
+    // refresh the account list
+    gas_populate_list (gas);
+}
+
+static void
+toggle_placeholder_cb (GtkWidget *widget, gpointer user_data)
+{
+    GNCAccountSel *gas = GNC_ACCOUNT_SEL(user_data);
+    gas->hide_placeholder = gtk_check_menu_item_get_active (GTK_CHECK_MENU_ITEM(widget));
+    update_entry_and_list (gas);
+}
+
+static void
+toggle_hidden_cb (GtkWidget *widget, gpointer user_data)
+{
+    GNCAccountSel *gas = GNC_ACCOUNT_SEL(user_data);
+    gas->hide_hidden = gtk_check_menu_item_get_active (GTK_CHECK_MENU_ITEM(widget));
+    update_entry_and_list (gas);
+}
+
+static void
+icon_release_cb (GtkEntry *entry, GtkEntryIconPosition icon_pos,
+                 GdkEvent* event, gpointer user_data)
+{
+    GNCAccountSel *gas = GNC_ACCOUNT_SEL(user_data);
+    GtkWidget *menu, *h_placeholder, *h_hidden;
+
+    if (icon_pos != GTK_ENTRY_ICON_SECONDARY)
+        return;
+
+    menu = gtk_menu_new ();
+    h_placeholder = gtk_check_menu_item_new_with_label (_("Hide Placeholder Accounts"));
+    gtk_check_menu_item_set_active (GTK_CHECK_MENU_ITEM(h_placeholder), gas->hide_placeholder);
+    h_hidden = gtk_check_menu_item_new_with_label (_("Hide Hidden Accounts"));
```

Comment:
Why not adding accellerators?
"Hide _Placeholder Accounts"
"Hide _Hidden Accounts"

Reply:
Did not think of it, added